JoomGalleryfriends/JG4-dev
# Overhaul of FileManager service

## What is this service for

The FileManager service will be used to handle the files and folders created in the filesystem. I offers methods to perform processes (create, delete, check, move) on image files and folders.

**This is a overhaul of PR #26.**

The usage of the FileManager is exactly the same as before but now with a complete new filesystem service. The new filesystem service support the filesystem adapter provider of com_media. Meaning that different filesystems can be used based on which filesystem plugins you have installed.

## What are filesystem plugins

Filesystem plugins are a new type of plugins introduced with Joomla 4. They allow Joomla to communicate with different filesystems. In the Joomla core only one filesystem plugin is preinstalled to do operations with files on the local hosted filesystem:
![grafik](https://user-images.githubusercontent.com/39154009/226292149-b00ca48c-0054-4930-95ab-a09365754326.png)

There are already several additional filesystem plugins available allowing you to operate with files on other, even not locally hosted filesystems like gDrive, OneCloud, Amazon S3, ...
- https://extensions.joomla.org/extension/amazon-s3-filesystem/
- https://extensions.joomla.org/extension/dpmedia/
- https://www.joomlacontenteditor.net/downloads/plugins/server-filesystem
- https://extensions.joomla.org/extension/google-cloud-storage/

This results that JoomGallery start to support multiple filesystems with this PR without the need of writing code by ourselfs.

## How to use this service

See PR #26

## How to test this service

See PR #26. Everything should work the same as before. You can even use the operations in the GUI to perform the tests.

Affected opertaions:
- create Images
- delete Images
- check Images
- move Images
- copy Images
- create Category
- delete Category
- check Category
- move Category
- copy Category
- get image path
- get category path
